### ISSUE TITLE
Capitalization in variable names not preserved in some cases

### ISSUE THREAD

**Issue by justaparth** [vojtechhabarta/typescript-generator#357]
Hello,

I found that if a private instance variable with a standard getter (generated by Lombok or by hand) starts with a single lowercase letter followed by a capital letter, the capitalization is not preserved correctly. 

I've made a reproducible and runnable version of my code here: https://github.com/justaparth/typescriptgentest

```
@Value
class TestClassLombok {
    int xPosition;
    int aLongerVariableName;
    int abPosition;
}
```

produces:

```
interface TestClassLombok {
    abPosition: number;
    xposition: number;
    alongerVariableName: number;
}
```

where I would have expected:

```
interface TestClassLombok {
    abPosition: number;
    xPosition: number;
    aLongerVariableName: number;
}
```

I suspect it is related to the getter somehow, since if I don't use lombok and define the getter as `getxPosition` instead of `getXPosition`, then the capitalization works: (https://github.com/justaparth/typescriptgentest/blob/master/src/main/java/com/parth/testing/Classes.java#L27). 

Any help would be appreciated, thanks!

**Comment by vojtechhabarta** [comment]
Hello, thanks for the test repo.
I added `main` method which serializes `TestClassLombok` instance to JSON:

``` java
public static void main(String[] args) throws Exception {
    final ObjectMapper objectMapper = new ObjectMapper();
    objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
    System.out.println(objectMapper.writeValueAsString(new TestClassLombok(10, 10, 10)));
}
```
and the output was:
``` json
{
  "abPosition" : 10,
  "alongerVariableName" : 10,
  "xposition" : 10
}
```
so it seems that typescript-generator handles these names correctly.
It gets list of properties from Jackson library so it should be consistent with real JSON data.

This naming depends on Jackson cooperation with Lombok.

**Comment by vojtechhabarta** [comment]
@justaparth were you able to make it work as needed?

**Comment by justaparth** [comment]
@vojtechhabarta Sorry for the late response, and thanks for the clarification. Just as you said, I confirmed it was consistent with the JSON serialization, which makes sense as the behavior typescript generator would want.

Just in case anyone else comes across this, it seems at least some other people have had this issue with Lombok-generated getters for variable names of this type, i.e. `xPosition`'s getter ends up becoming `getXPosition` instead of `getxPosition`. But it seems that various implementations do different things, so its a bit of a gray area :( (https://github.com/rzwitserloot/lombok/issues/1661).

In any case, after reading your comment, I was able to resolve it by not using lombok getters and defining them myself, getting consistent variable names and json serialization. Thanks again for your help!

```
class TestClassLombok {
    int xPosition;
    int aLongerVariableName;
    int abPosition;
    public int getxPosition() { .... }
    ....
}
```
